CiscoCloud/mantl
# kafka: create topics and cleanup config

- [x] Installs cleanly on a fresh build of most recent master branch
- [x] Upgrades cleanly from the most recent release
- [x] Updates documentation relevant to the changes

---

* updates default broker config
* auto-creates the specified topics
* easier to understand config
* sets advertised.host.name as the consul service name for the broker
* optional installation of kafka manager: https://github.com/yahoo/kafka-manager
* uses a new consul acl token for writing to consul during framework addon installs (requires re-running security-setup)

---

Testing:

* install the kafka addon (`ansible-playbook -e @security.yml addons/kafka.yml`)
* verify that the there are 3 brokers started after a few minutes
* broker-1, broker-2, and broker-3 should be registered in consul
* verify that the 4 topics were created. You can ask the kafka api by running a command like:

`curl -sku admin:pw https://control-01/kafka/api/topic/list`

* optionally install kafka-manager (`ansible-playbook -e @security.yml -e 'kafka_manager_install=yes' addons/kafka.yml`)
  * to test the kafka manager ui, you can either add `kafka_manager_load_balancer=external` when installing to expose it via traefik (assuming you have dns) or find the host and port in marathon and then open firewall/security group accordingly
